printTables.py

```diff
@@ -0,0 +1,247 @@
+
+
+    def __init__(self, matrixTable, maxRowWidth):
+        self.table = matrixTable
+        self.boarder = "|"
+        self.rowSym = "-"
+        self.intersectionChar = "+"
+        self.outputString = ""
+        self.tableSize = len(self.table)
+        self.headerSize = 0
+        self.getHeaderSize()
+        self.maxWordLen = int(maxRowWidth / self.headerSize)
+        self.maxRowWidth = maxRowWidth
+        self.columnWidth = []
+
+
+
+
+
+    def getHeaderSize(self):
+
+        if self.tableSize > 0:
+            self.headerSize = len(self.table[0])
+
+
+    """
+    Accessor function for the maximum word length variable 
+    """
+    def getMaxWordLength(self):
+
+        return self.maxWordLen
+
+    """
+    Mutator function maximum word length  variable 
+    """
+    def setMaxWordLength(self, newLen):
+
+        self.maxWordLen = newLen
+
+    """
+     Accessor function for the boarder character variable
+    """
+    def getBoarderChar(self):
+
+        return str(self.boarder)
+
+    """
+    Mutator function for the boarder character variable 
+    """
+    def setBoarderChar(self, newChar):
+
+        self.boarder = str(newChar)
+
+    def makeString(self, appendString):
+
+        self.outputString += appendString
+
+    """
+     Function is designed to multiply the member variable self.boarder to a specified number 
+     This function helps create the character designed boarder
+    """
+    def mulWordWithrowChar(self, num):
+
+        return str(self.rowSym * num)
+
+    """
+    Function calculates the distance for where the last self.boarder  needs to be placed when printing a row. 
+    """
+    def finalSymError (self):
+
+        return self.maxWordLen - 18
+
+    """
+    Function calculates the number of self.boarder chars needed to align perfectly, when printing a solid lin 
+    of all self.boarder chars. This is useful when the self.maxWordLen changes and the table boarders need to be 
+    resized. 
+    """
+    def resizing (self):
+
+        return self.maxWordLen * 5 - 3
+
+    """
+     Function obtains the longest world length in a matrix table 
+     Function could be used to make a more dynamic, organic word length and set it to 
+     the class member world bound self.maxWordLen 
+    """
+    def getGreatestWordLength (self):
+
+        length = 0
+
+        for i in range(1, len(self.table)):
+            for j in range(len(self.table[i])):
+                if length < len(str(self.table[i][j])):
+                    length = len(str(self.table[i][j]))
+
+        return length
+
+
+    """
+    Function is designed to take in an integer in the n parameter. Find the longest length of any string a matrix 
+    array and add n to it. Then set this value to the class member self.maxWordLen.  In so doing, you can organically 
+    create a column width to fit the needs of any table that is of a matrix 2D array type of data structure. 
+    """
+    def organicCOLCalculation(self, n):
```

Comment:
This function name should be more clear. Why is it 'organic'?

Reply:
Because it is supposed to loop though and find the longest word in a table. For any table type passed to it. Therefore, organic Column calculation means find the next longest work in a table passed and recalculate it on the  fly.  To bad the rest of the mathematics was elementary at best. Otherwise this theoretically could have worked out.